File: app/src/main/java/com/kite/mnemoai/viewmodels/StatisticViewModel.java
package com.kite.mnemoai.viewmodels;

import static androidx.lifecycle.ViewModelProvider.AndroidViewModelFactory.APPLICATION_KEY;

import androidx.lifecycle.ViewModel;
import androidx.lifecycle.viewmodel.ViewModelInitializer;

import com.kite.mnemoai.MainApplication;
import com.kite.mnemoai.data.local.DTO.StudyStatistic;
import com.kite.mnemoai.data.repository.IRepositoryCallback;
import com.kite.mnemoai.data.repository.StatisticsRepository;

import java.util.List;

public class StatisticViewModel extends ViewModel {
    private StatisticsRepository statisticsRepository;

    public StatisticViewModel(StatisticsRepository statisticsRepository) {
        this.statisticsRepository = statisticsRepository;
    }

    public void getStudyStatistic(IRepositoryCallback<List<StudyStatistic>> callback){
        statisticsRepository.getStudyStatistic(callback);
    }

    public final static ViewModelInitializer<StatisticViewModel> initializer = new ViewModelInitializer<>(
            StatisticViewModel.class,
            creationExtras -> {
                MainApplication app = (MainApplication) creationExtras.get(APPLICATION_KEY);
                assert app != null;
                return new StatisticViewModel(app.getStatisticsRepository());
            }
    );

}

File: app/src/main/java/com/kite/mnemoai/ui/statistic/StatisticViewModel.kt
package com.kite.mnemoai.ui.statistic

import androidx.lifecycle.ViewModel
import com.kite.mnemoai.data.model.StudyStatistic
import com.kite.mnemoai.data.repository.IRepositoryCallback
import com.kite.mnemoai.data.repository.StatisticsRepository
import dagger.hilt.android.lifecycle.HiltViewModel
import javax.inject.Inject

@HiltViewModel
class StatisticViewModel @Inject constructor(
    private val statisticsRepository: StatisticsRepository
) : ViewModel() {
    fun getStudyStatistic(callback: IRepositoryCallback<MutableList<StudyStatistic?>?>?) {
        statisticsRepository.getStudyStatistic(callback)
    }
}
